helios-tools/src/main/java/com/spotify/helios/auth/BasicClientAuthenticationPlugin.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.auth;
+
+import com.spotify.helios.client.RequestDispatcher;
+
+public class BasicClientAuthenticationPlugin implements ClientAuthenticationPlugin {
```

Comment:
Class names could be shorter here. Perhaps just call it `AuthenticationPlugin` on both the server and client and let the name be overloaded?